lib/Doctrine/DBAL/Types/DateTimeType.php

```diff
@@ -66,7 +66,7 @@ public function convertToDatabaseValue($value, AbstractPlatform $platform)
      */
     public function convertToPHPValue($value, AbstractPlatform $platform)
     {
-        if ($value === null || $value instanceof \DateTimeInterface) {
```

Comment:
I think this is still needed in case of `DateTime`, because as far as I remember PDO Sqlite or sqlsrv immediately return DateTime objects for date/time/datetime typed fields.

Reply:
If that's was the case, we'd most likely have failures in `TypeConversionTest` with those drivers. However, double checking won't hurt.